v2/blockstore/readwrite_test.go

```diff
@@ -291,25 +291,62 @@ func TestBlockstoreResumption(t *testing.T) {
 	require.NoError(t, err)
 
 	// For each block resume on the same file, putting blocks one at a time.
+	var wantBlockCountSoFar int
+	wantBlocks := make(map[cid.Cid]blocks.Block)
 	for {
 		b, err := r.Next()
 		if err == io.EOF {
 			break
 		}
 		require.NoError(t, err)
-
-		// 30% chance of subject failing (or closing file without calling Finalize).
-		// The higher this percentage the slower the test runs considering the number of blocks in the original CAR v1 test payload.
-		shouldFailAbruptly := rand.Float32() <= 0.3
-		if shouldFailAbruptly {
-			// Close off the open file and re-instantiate a new subject with resumption enabled.
-			// Note, we don't have to close the file for resumption to work.
-			// We do this to avoid resource leak during testing.
-			require.NoError(t, subject.Close())
+		wantBlockCountSoFar++
+		wantBlocks[b.Cid()] = b
+
+		// 30% chance of subject failing; more concretely: re-instantiating blockstore with the same
+		// file without calling Finalize. The higher this percentage the slower the test runs
+		// considering the number of blocks in the original CAR v1 test payload.
+		resume := rand.Float32() <= 0.3
```

Comment:
we should make tests deterministic

Reply:
@mvdan does go use a fixed seed for rand by default in these test cases?

Reply:
Would a `rand` with fixed seed suffice?

Reply:
> @mvdan does go use a fixed seed for rand by default in these test cases?

I _think_ this is a yes.

Reply:
This code is using the global source, so you can't know for sure. It's likely that some dependency somewhere is doing something akin to `func init() { rand.Seed(...) }`.

So yeah, use your custom source with a fixed seed.